cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/TranslationConfiguration.kt

```diff
@@ -453,6 +456,20 @@ private constructor(
             return this.configurePass(T::class, config)
         }
 
+        /**
+         * Sets a list of exclusion patterns for filtering files and directories.
+         *
+         * @param patterns List of exclusion patterns: strings are treated as Regex patterns.
+         *   Example:
+         * ```
+         * exclusionPatterns(listOf(".*test(s)?", "tests"))
+         * ```
+         */
+        fun exclusionPatterns(patterns: List<String>): Builder {
```

Comment:
Any particular reason not just expose this as a List<Regex>? Seems error-prone if I expect a list that is somehow turned into a regex magically.

Reply:
Done, added separate functions for regex and strings